benchmark/Main.hs

```diff
@@ -167,3 +169,11 @@ bgroupMember = do
                     , bench "HashSet/Foldable" $ whnf (Foldable.elem  sample) listHashSet
                     , bench "HashSet/elem"     $ whnf (Container.elem sample) listHashSet
                     ]
+
+-- | Checks that 'foldl'' is implemented efficiently for 'Universum.List'
+bgroupFold :: Benchmark
+bgroupFold = do
+    let testList    = [1..100000] :: [Int]
+    let foo         = foldl' HashSet.insert mempty
+    bgroup "fold"   [ bench "List/foldl'" $ nf foo testList
+                    ]
```

Comment:
It's not enough to see time of our `foldl'` implementation because time is relative. And you only can tell something in comparison with other functions. I propose to have `ghcFoldl'` which uses `foldl'` from `base` package. Like this:

```haskell
import qualified GHC.List as List (foldl')
let ghcFoldl' = List.foldl' (\hashSet element -> HashSet.insert element hashSet) mempty
```

and then

```haskell
bgroup "fold"   [ bench "List/foldl'" $ nf universumFoldl' testList
                , bench "Base/foldl'" $ nf ghcFoldl' testList
                ]
```